wa/instruments/perf.py

```diff
@@ -123,16 +170,134 @@ def update_output(self, context):
                             in_results_section = False
                             break
                         else:
-                            line = line.split('#')[0]  # comment
-                            match = PERF_COUNT_REGEX.search(line)
-                            if match:
-                                classifiers = {}
-                                cpu = match.group(1)
-                                if cpu is not None:
-                                    classifiers['cpu'] = int(cpu.replace('CPU', ''))
-                                count = int(match.group(2))
-                                metric = '{}_{}'.format(label, match.group(3))
-                                context.add_metric(metric, count, classifiers=classifiers)
+                            self._add_perf_stat_metric(line, label, context)
 
-    def teardown(self, context):
-        self.collector.reset()
+    @staticmethod
+    def _add_perf_stat_metric(line, label, context):
+        line = line.split('#')[0]  # comment
+        match = PERF_COUNT_REGEX.search(line)
+        if match:
```

Comment:
With checks such as this, you can get rid of an indent level by reversing the logic:

```
if not match:
     return   # or "continue" inside a loop
# the rest of the code does not need to be indented after "if".
```